[ADD] marketing automation: double opt-in template (#9513)
odoo/documentation: content/applications/marketing/marketing_automation/campaign_templates/double_optin.rst

click on a link in the email are automatically added to the *Confirmed contacts* mailing list.

.. important::
   When using the *Double Opt-in* campaign template, only the contacts in the *Confirmed contacts
   mailing* list are considered to have confirmed their consent.

.. _marketing_automation/template/using-double-optin:

Use the Double Opt-in campaign template
=======================================

Open the :menuselection:`Marketing Automation` app, and select the :guilabel:`Double Opt-in`
campaign template to create a new campaign for confirming consent.

.. tip::
   The campaign templates do not display by default when there are existing *Marketing Automation*
   campaigns. To display the campaign templates, type the name of a campaign that does not exist in
   the database into the :guilabel:`Search...` bar, then press :kbd:`Enter`.

   For example, searching for `empty` displays the campaign template cards again, as long as there
   is not a campaign with the name "empty" in the database.

Campaign configuration
----------------------

Upon creation of the campaign, the campaign form loads with a new preconfigured campaign.

> The :guilabel:`Target` and :guilabel:`Filter` configuration of the campaign is as follows:

Reviewer: ```suggestion
The :guilabel:`Target` and :guilabel:`Filter` configurations of the campaign are as follows:
```

- :guilabel:`Name`: `Double Opt-in`

> - :guilabel:`Responsible`: The user who created the campaign.

Reviewer: I'm actually not seeing a 'Responsible' field on the 17.0 runbot I'm using for this review --- but that could be a RunBot issue haha -- worth a double-check, either way, though...
![Screenshot 2024-05-29 110228](https://github.com/odoo/documentation/assets/73958186/47b75e0d-5378-4dba-a4a3-fa30a1e101d9)

Author: The Responsible field only shows with developer mode active, I will make a note :)

- :guilabel:`Target`: :guilabel:`Mailing Contact`
- :guilabel:`Unicity based on`: :guilabel:`Email (Mailing Contact)`
- :guilabel:`Filter`:

  - :guilabel:`Email` :guilabel:`is set`
  - :guilabel:`Blacklist` :guilabel:`is not` :guilabel:`set`

>   - :guilabel:`Mailing lists` :guilabel:`contains` `Newsletter`

Reviewer: ```suggestion
  - :guilabel:`Mailing lists` :guilabel:`contains` :guilabel:`Newsletter`
```
just to follow the pattern established by you in the previous 2 lines 👍

Author: This should be literal formatting as the user can type in the field

.. important::

>    The :guilabel:`Target` model of the campaign should not be modified. Changing the

Reviewer: ```suggestion
   The :guilabel:`Target` model of the campaign should **not** be modified. Changing the
```

   :guilabel:`Target` model with activities in the :guilabel:`Workflow` invalidates the existing
   activities in the :guilabel:`Workflow`.

   The *Double Opt-in* campaign template is intended to only use the :guilabel:`Mailing Contact`
   model.

The campaign loads two activities in the :guilabel:`Workflow` section of the campaign: an email

> activity with a child server action activity that triggers *on click*.

Reviewer: ```suggestion
activity, with a child server action activity that triggers *on click*.
```

> By default, the `Confirmation` email activity is set to trigger :guilabel:`1 Hours` after the
> beginning of the workflow. In other words, the email is sent 1 hour after a new contact is added to
> the *Newsletter* mailing list.

Reviewer: ```suggestion
By default, the :guilabel:`Confirmation` email activity is set to trigger :guilabel:`0 Hours` after the
beginning of the workflow. In other words, the email is sent right after a new contact is added to
the *Newsletter* mailing list.
```
I'm suggesting the addition of a GUI because it actually *appears* on-screen, but I am actually seeing that it is set to '0' hours, by default -- but again, that may be a 'runbot' specific discrepancy. worth a double-check during your revisions either way.
![Screenshot 2024-05-29 110715](https://github.com/odoo/documentation/assets/73958186/738c8270-5a31-421d-a041-3aaeda56a112)

The email activity uses the preconfigured *Confirmation* email template, which contains a button for
the contact to click to confirm their consent.

To modify the email template, select the :icon:`fa-envelope-o` :guilabel:`Templates` smart button at

> the top of the campaign form. Then, in the list of templates, select the `Confirmation` email

Reviewer: ```suggestion
the top of the campaign form. Then, in the list of templates, select the :guilabel:`Confirmation` email
```

Author: Since this is the title of the activity, which can be altered by the user typing, I am leaving it as literal formatting :)

template.

Be sure to personalize the contents of the email template; however, it is recommended to keep the

> contents of double opt-in confirmation emails short and to-the-point.

Reviewer: suggestion: would the "however, it is recommended..." portion be better suited for a tip admonition? Might help it stand out a bit more, too. Just a thought...

The default confirmation button links directly to the database's website homepage. Click on the
button to edit the button text and URL.

.. tip::

>    To provide a streamlined user experience for the contact, consider :doc:`creating a page on the

Reviewer: ```suggestion
   To provide a streamlined experience for the contact, consider :doc:`creating a page on the
```
I've been instructed by ZST in recent PRs to avoid using the term "user" -- hence, my suggestion here

   website <../../../websites/website/pages>` that expresses gratitude to the contact for
   confirming their subscription to the mailing list. Add the link to that page in the URL of the
   confirmation button.

.. important::

>    The email template should only include a single call to action link for confirmation, other than

Reviewer: ```suggestion
   The email template should only include a single call-to-action link for confirmation, other than
```

   an unsubscribe link.

>    Any click on a link or button included in the confirmation email, besides the unsubscribe button,

Reviewer: ```suggestion
   Any click on a link (or button) included in the confirmation email, besides the unsubscribe button,
```

   triggers the *Add to list* server action.

   The child activity *Add to list* server action's *On click* trigger cannot differentiate between
   multiple URLs in an email, besides the `/unsubscribe_from_list` unsubscribe button that is
   included in any one of the footer blocks.

> The `Add to list` server action child activity is set to trigger immediately after a click in the

Reviewer: ```suggestion
The :guilabel:`Add to list` server action child activity is set to trigger immediately after a click in the
```
suggesting a GUI here because, at this point, the reader would be back on the campaign form page, wherein this would *actually be seen* on the screen

parent email activity is detected. When triggered, the server action activity executes the
preconfigured *Add To Confirmed List* server action, which adds the contact to the *Confirmed
contacts* mailing list, if they are not already in the mailing list.

To modify the server action, select the title of the activity to open the :guilabel:`Open:
Activities` pop-up window and edit the server action activities configuration.

.. tip::
   Consider setting an :guilabel:`Expiry Duration` to prevent executing the activity after a
   specific amount of time.

.. important::
   It is not recommended to modify the preconfigured Python code in the :guilabel:`Add To Confirmed
   List` server action, as doing so may trigger a change in the database's pricing plan.

Once the campaign configuration is complete, consider :doc:`launching a test <../testing_running>`
to verify the campaign executes as expected. If the campaign testing is successful,
:guilabel:`Start` the campaign to begin sending double opt-in confirmation emails to *Newsletter*
mailing list contacts, and fill the *Confirmed contacts* mailing list with engaged contacts.

.. _marketing_automation/template/double-optin-usecase:

> Double opt-in use-case

Reviewer: ```suggestion
Double Opt-in use-case
```
same suggestion re: capitalization as mentioned above

======================

.. example::
   To prepare for sending newsletter marketing emails on an Odoo database, a mailing contact list
   must be procured. One way of collecting subscribers is through a sign-up form on the website that
   adds contacts to the *Newsletter* mailing list on form submission.

   .. image:: double_optin/newsletter-signup.png
      :align: center
      :alt: Newsletter sign-up form on Odoo website footer.

   Before sending any marketing emails, :ref:`use the Double Opt-in campaign template
   <marketing_automation/template/using-double-optin>` in the *Marketing Automation* app to confirm
   marketing email consent from the contacts in the *Newsletter* mailing list.

   After launching the *Double Opt-in* campaign, view the contacts that have double opt-in in the

>    *Confirmed contacts* mailing list (:menuselection:`Email Marketing --> Mailing Lists --> Mailing

Reviewer: ```suggestion
   *Confirmed contacts* mailing list (:menuselection:`Email Marketing app --> Mailing Lists --> Mailing
```

   Lists`).

   .. image:: double_optin/double-optin-metrics.png
      :align: center

>       :alt: Newsletter sign-up form on Odoo website footer.

Author: Note for myself to update this alt tag
```suggestion
      :alt: Activity metrics on the campaign form.
```

>    Now the *Confirmed contacts* mailing list is ready to be used for sending newsletter marketing

Reviewer: ```suggestion
   Now, the *Confirmed contacts* mailing list is ready to be used for sending newsletter marketing
```

Open the :menuselection:`Marketing Automation` app, and select the :guilabel:`Double Opt-in`
campaign template to create a new campaign for confirming consent.

.. tip::
   The campaign templates do **not** display, by default, when there are existing *Marketing
   Automation* campaigns. To display the campaign templates, type the name of a campaign (that does
   not exist in the database) into the :guilabel:`Search...` bar, then press :kbd:`Enter`.

   For example, searching for `empty` displays the campaign template cards again, as long as there
   is not a campaign with the name "empty" in the database.

Campaign configuration
----------------------

Upon creation of the campaign, the campaign form loads with a new preconfigured campaign.

The :guilabel:`Target` and :guilabel:`Filter` configurations of the campaign are as follows:

- :guilabel:`Name`: `Double Opt-in`
- :guilabel:`Responsible`\*: The user who created the campaign.
- :guilabel:`Target`: :guilabel:`Mailing Contact`
- :guilabel:`Unicity based on`: :guilabel:`Email (Mailing Contact)`
- :guilabel:`Filter`:

  - :guilabel:`Email` :guilabel:`is set`
  - :guilabel:`Blacklist` :guilabel:`is not` :guilabel:`set`

>   - :guilabel:`Mailing lists` :guilabel:`contains` `Newsletter`

Reviewer 2: Alternative formatting for filters: https://www.odoo.com/documentation/17.0/applications/marketing/email_marketing/lost_leads_email.html

website homepage. Click on the button to edit the button text and URL.

.. tip::
   To provide a streamlined experience for the contact, consider :doc:`creating a page on the
   website <../../../websites/website/pages>` that expresses gratitude to the contact for
   confirming their subscription to the mailing list. Add the link to that page in the URL of the
   confirmation button.

.. important::
   The email template should only include a single call-to-action link for confirmation, other than
   an unsubscribe link.

   Any click on a link (or button) included in the confirmation email, besides the unsubscribe
   button, triggers the *Add to list* server action.

   The child activity *Add to list* server action's *On click* trigger cannot differentiate between
   multiple URLs in an email, besides the `/unsubscribe_from_list` unsubscribe button that is
   included in any one of the footer blocks.

The `Add to list` server action child activity is set to trigger immediately after a click in the
parent email activity is detected. When triggered, the server action activity executes the
preconfigured *Add To Confirmed List* server action, which adds the contact to the *Confirmed

> contacts* mailing list, if they are not already in the mailing list.

Reviewer 2: Can condense down with clearer language.
